packages/web-react/src/hooks/useDeprecatedMessage.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+
+export interface UseDeprecatedMessageProps {
+  trigger?: boolean;
+  deprecatedPropName?: string;
+  newPropName?: string;
+}
+
+export const useDeprecatedMessage = ({ trigger, deprecatedPropName, newPropName }: UseDeprecatedMessageProps): void => {
+  useEffect(() => {
+    if (trigger && deprecatedPropName && newPropName) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Deprecated warning: "${deprecatedPropName}" property will be replaced soon, use "${newPropName}" instead.`,
+      );
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Here should be `trigger`, `deprecatedPropName`, `newPropsName`. Eg. the effect will not trigger when the variable does not change. In this case it will trigger every time. Is this intention?

Reply:
nope, run only once